app/models/category.rb

```diff
@@ -0,0 +1,6 @@
+class Category < ApplicationRecord
+  has_and_belongs_to_many :products
+
+  validates :name, presence: true, uniqueness: true
```

Comment:
I think `case_sensitive: false` will be more suitable for category name.
```suggestion
  validates :name, presence: true, uniqueness: { case_sensitive: false }
```

Reply:
Fixed it bhai. Thank you.